server/service/client.go

```diff
@@ -658,6 +658,10 @@ func (c *Client) ApplyGroup(
 				}
 				if b, ok := tmMacSetupAssistants[tmName]; ok {
 					if err := c.uploadMacOSSetupAssistant(b, &tmID, tmMacSetup[tmName].MacOSSetupAssistant.Value); err != nil {
+						if strings.Contains(err.Error(), "Couldn't upload") {
+							parts := strings.Split(err.Error(), ".")
+							return nil, nil, fmt.Errorf("Couldn't edit macos_setup_assistant. Response from Apple: %s. Learn more at %s", strings.Trim(parts[1], " "), "https://fleetdm.com/learn-more-about/dep-profile")
```

Comment:
ouch, there's a potential panic here. I do understand that this is a tricky situation tho. If we don't have any tests that cover this, could you add a sanity check fallback?